src/modules/reviewrecords.js

```diff
@@ -107,19 +127,21 @@ class ReviewRecords {
             publish = true,
             validate = true
         } = options;
-        const buffer = await this.prepareReviewRecord(reviewRecord, validate);
+        const rr = await this.prepareReviewRecord(reviewRecord, validate);
+        const buffer = protobuf.ReviewRecord.encode(rr);
         const dagNode = await this.chluIpfs.ipfsUtils.createDAGNode(buffer); // don't store to IPFS yet
         const multihash = IPFSUtils.getDAGNodeMultihash(dagNode);
         if (options.expectedMultihash) {
             if (options.expectedMultihash !== multihash) {
                 throw new Error('Expected a different multihash');
             }
         }
-        if (publish) await this.publishReviewRecord(dagNode, previousVersionMultihash, multihash);
+        this.chluIpfs.events.emit('stored ReviewRecord', { multihash, reviewRecord: rr });
+        if (publish) await this.publishReviewRecord(dagNode, previousVersionMultihash, multihash, rr);
         return multihash;
     }
 
-    async publishReviewRecord(dagNode, previousVersionMultihash, expectedMultihash) {
+    async publishReviewRecord(dagNode, previousVersionMultihash, expectedMultihash, reviewRecord) {
```

Comment:
How do we handle the situation that waiting for remote pin never returns?

Reply:
At the moment we don't. Should we put a timeout?

Reply:
Ideally we would time out and communicate the error to user. We can create a new issue for post-demo for this.